projectgenerator/gui/multiple_models.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+"""
+/***************************************************************************
+                              -------------------
+        begin                : 25.11.2017
+        git sha              : :%H$
+        copyright            : (C) 2017 by Germán Carrillo (BSF-Swissphoto)
+        email                : gcarrillo@linuxmail.org
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+"""
+from projectgenerator.utils import get_ui_class
+
+from qgis.PyQt.QtCore import Qt
+from qgis.PyQt.QtWidgets import QDialog, QListWidgetItem, QDialogButtonBox
+
+
+DIALOG_UI = get_ui_class('multiple_models.ui')
+
+class MultipleModelsDialog(QDialog, DIALOG_UI):
+    def __init__(self, parent=None):
+        QDialog.__init__(self, parent)
+        self.setupUi(self)
+        self.parent = parent
+
+        self.model_list.itemSelectionChanged.connect(self.on_selection_changed)
+        self.on_selection_changed()
+
+        self.add_button.clicked.connect(self.add_model)
+        self.remove_button.clicked.connect(self.remove_model)
+        self.buttonBox.accepted.connect(self.accepted)
+
+        self.models_line_edit.textChanged.connect(self.update_add_button_state)
+        self.models_line_edit.textChanged.emit(self.models_line_edit.text())
+
+    def add_model(self):
+        model_name = self.models_line_edit.text().strip()
+        if not self.model_list.findItems(model_name, Qt.MatchExactly):
+            self.model_list.addItem(QListWidgetItem(model_name))
+            self.models_line_edit.setText("")
+
+    def remove_model(self):
+        for item in self.model_list.selectedItems():
+            self.model_list.takeItem(self.model_list.row(item))
+
+    def accepted(self):
+        items = [self.model_list.item(x) for x in range(self.model_list.count())]
+        models = ";".join([i.text().strip() for i in items if i.text().strip()])
+        self.parent.ili_models_line_edit.setText(models)
```

Comment:
What do you think about changing this to the approach to connect the parent to the dialog's accepted signal instead of pushing the model from here. Then just have a very simple method in here to generate the model string

    def model_string(self):
        items = [self.model_list.item(x) for x in range(self.model_list.count())]
        models = ";".join([i.text().strip() for i in items if i.text().strip()])
        return models

This would reduce the assumption that this dialog has to make about the environment.
No big deal though ;)

Reply:
I think it makes sense. I've just adjusted the PR.